File: C++/Easy/Accepted/count-binary-substrings_2.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        
        int res = 0;
        int pre = 0, cur = 1;
        for (int i = 1; i < s.size(); ++i) {
            if (s[i] == s[i - 1]) {
                ++cur;
            } else {
                res += min(pre, cur);
                pre = cur;
                cur = 1;
            }
        }
        return res + min(pre, cur);
    }
};

File: Java/Easy/Accepted/count-binary-substrings_2.java
class Solution {
    public int countBinarySubstrings(String s) {
        
        int res = 0;
        int prev = 0;
        int cur = 1;
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == s.charAt(i - 1)) {
                cur++;
            } else {
                prev = cur;
                cur = 1;
            }
            if (prev >= cur) {
                res++;
            }
        }
        return res;
    }
}